src/nvim/os/win_defs.h

```diff
@@ -6,16 +6,22 @@
 #define TEMP_DIR_NAMES {"$TMP", "$TEMP", "$USERPROFILE", ""}
 #define TEMP_FILE_PATH_MAXLEN _MAX_PATH
 
-// Defines needed to fix the build on Windows:
-// - DFLT_DIR
-// - DFLT_BDIR
-// - DFLT_VDIR
-// - EXRC_FILE
-// - VIMRC_FILE
-// - SYNTAX_FNAME
-// - DFLT_HELPFILE
-// - SYS_VIMRC_FILE
-// - SPECIAL_WILDCHAR
+// Windows system-dependent file names
+#ifndef SYS_VIMRC_FILE
+# define SYS_VIMRC_FILE "$VIM\\sysinit.vim"
+#endif
+#ifndef DFLT_HELPFILE
+# define DFLT_HELPFILE  "$VIMRUMTIME\\doc\\help.txt"
+#endif
+#ifndef SYNTAX_FNAME
+# define SYNTAX_FNAME   "$VIMRUNTIME\\syntax\\%s.vim"
```

Comment:
Yes BACKSLASH_IN_FILENAME is only set for Windows, but PATHSEPSTR depends on `shellslash` (pseps) defined in globals.h. We can also leave PATHSEPSTR for later since we will bump into it again in f7e590f4345ae4ddd86db3f196140c80535fcc2b